### ISSUE TITLE
Segfault on debian buster + sway

### ISSUE THREAD

**Issue by spaceone** [atx/wtype#10]
I got a segfault on executing:
```
$ gdb --args wtype test
(gdb) run
Starting program: /usr/local/bin/wtype test
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00005555555564b8 in main ()
(gdb) bt
#0  0x00005555555564b8 in main ()
```

Here is my installation log:
```
$ meson build                                                                                                                                                                                                
The Meson build system
Version: 0.49.2
Source dir: /home//git/wtype
Build dir: /home//git/wtype/build
Build type: native build
Project name: wtype
Project version: 0.0.1
Native C compiler: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Program git found: YES (/usr/bin/git)
Found pkg-config: /usr/bin/pkg-config (0.29)
Dependency wayland-client found: YES 1.16.0
Dependency wayland-cursor found: YES 1.16.0
Dependency xkbcommon found: YES 0.8.2
Library rt found: YES
Program wayland-scanner found: YES (/usr/bin/wayland-scanner)
Build targets in project: 2
Found ninja-1.8$ ninja -C build
ninja: Entering directory `build'
[6/7] Compiling C object 'wtype@exe/main.c.o'.
../main.c: In function ‘get_key_code’:
../main.c:136:18: warning: implicit declaration of function ‘reallocarray’; did you mean ‘realloc’? [-Wimplicit-function-declaration]
  wtype->keymap = reallocarray(
                  ^~~~~~~~~~~~
                  realloc
../main.c:136:16: warning: assignment to ‘wchar_t *’ {aka ‘int *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  wtype->keymap = reallocarray(
                ^
[7/7] Linking target wtype.
.2 at /usr/bin/ninja
$ su -c  "ninja -C build install"
Passwort:
ninja: Entering directory `build'
[0/1] Installing files.
Installing wtype to /usr/local/bin
```

**Comment by atx** [comment]
Ah right, looks like reallocarray needs _GNU_SOURCE on older glibcs. Should be fixed in 50d3436.

**Comment by spaceone** [comment]
Now `sway` crashes instead of getting a segfault. Where do I find wayland or sway logfiles?

**Comment by atx** [comment]
You can run sway with `-dV` to get a verbose debug output. If you are using a display manager, you probably have to consult its manual.

Which version of sway are you using? There does not seem to by any official buster package. Note that there have been some virtual-keyboard fixes in wlroots/sway semi-recently.

**Comment by spaceone** [comment]
```
$ sway --version
sway version 1.2 (Jun 18 2020, branch 'HEAD')
```

**Comment by atx** [comment]
That's way too old. There have been many virtual keyboard related patches both in sway and wlroots since then.

Try updating to the current (1.5) release (assuming it builds on buster).

**Comment by spaceone** [comment]
Is sway 1.4 good? I see the buster tutorial is nowerdays for 1.4 at https://github.com/swaywm/sway/wiki/Debian-10-(Buster)-Installation.

**Comment by atx** [comment]
Not sure, probably still too old. For instance, the bugs in #6 are fixed in 1.5 only.

**Comment by spaceone** [comment]
Ok, thank you.

I'll close this then and maybe re-check it when I upgrade sway (in ~1 year, seems there are version conflicts with packages in debian buster currently).
